### ISSUE TITLE
[Yarn/Lerna] Workspaces: Watcher not working in v2

### ISSUE THREAD

**Issue by krnlde** [parcel-bundler/parcel#6039]
# 🐛 bug report

## 🎛 Configuration (.babelrc, package.json, cli command)

* My whole configuration can be seen and cloned here https://github.com/lookmom/nohands/tree/feature/parcel2 ([pinned version](https://github.com/lookmom/nohands/tree/af0c2b2382a9c62d79f1925e034aea97c176b5c5))
* You'll need `node <=14` and `yarn <= 1.22.10`
* Install dependencies as usual with `yarn install`

## 🤔 Expected Behavior

when I start the project with `yarn start` and then change the [HelloWorld component](https://github.com/lookmom/nohands/blob/af0c2b2382a9c62d79f1925e034aea97c176b5c5/src/components/HelloWorld.tsx#L15) that is [imported](https://github.com/lookmom/nohands/blob/af0c2b2382a9c62d79f1925e034aea97c176b5c5/src/app/pages/Blog.tsx#L6) from a yarn workspace the project will not recompile. We've had numerous discussions about it. I reference them in the "context" part.

## 😯 Current Behavior

The Change is not recognized and the page stays pristine, when I change somethin inside the app workspace itself the watcher pics the changes up and recompiles. So that is not an issue.

## 💁 Possible Solution

Pick up changes from workspace imports like parcel v1 does.

## 🔦 Context

This same repo works in parcel v1. In fact you can reproduce the same steps in the [main ](https://github.com/lookmom/nohands)branch of the repo for comparison.
* Discussion about Lerna/ Yarn Workspaces: https://github.com/parcel-bundler/parcel/discussions/5429
* Possible duplicate issue, though not sure if: https://github.com/parcel-bundler/parcel/issues/4332
* Reply in an unrelated issue, re-assuring workspaces should be no problems: https://github.com/parcel-bundler/parcel/issues/5943#issuecomment-802347519

## 💻 Code Sample

https://github.com/lookmom/nohands/tree/feature/parcel2 ([pinned version](https://github.com/lookmom/nohands/tree/af0c2b2382a9c62d79f1925e034aea97c176b5c5))
## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | parcel@2.0.0-beta.2
| Node             | node@v15.12.0
| npm/Yarn         |  npm@7.6.3 , yarn@1.22.10
| Operating System | Windows 10 Pro (Insider Preview 21337)

The issue is present in every parcel v2 version, in every node version starting from 12, in every yarn version starting from 1.22, in every Windows 10 version starting from 1907. Also tested in WSL with Ubuntu 20.10 with the same results.

**Comment by krnlde** [comment]
Alright! I have more insights on this bug.

TL;DR: If there is a **false** `yarn.lock` inside the starter workspace, the `.parcel-cache` will be created there and as a result the Watcher does not work properly.

In my case I start the app via `parcel serve src/app/index.html` and I had an old `yarn.lock` inside `src/app`. Parcel v2 then creates a `src/app/.parcel-cache` and the Watcher won't work properly. If I then remove the `yarn.lock` and start parcel again the `.parcel-cache` will be created in the root folder (as it should) and the Watcher works.

Still a bug or an improvement maybe.

**Comment by mischnic** [comment]
Starting from your entry points, Parcel finds the first directory that contains a lockfile or a`.git` folder. This is the project root (any files outside of that aren't watched)

> Still a bug or an improvement maybe.

I'm not sure how this could be improved

**Comment by krnlde** [comment]
To clarify, I removed the dangling `src/app/yarn.lock`, nothing else, and it started working.

**Comment by krnlde** [comment]
Looks like this behaviour to me: https://github.com/parcel-bundler/parcel/blob/eabbbecf5a45981d2faa10be55d55115fc1c9bb4/packages/core/package-manager/src/installPackage.js#L137

**Comment by mischnic** [comment]
No, it's here: https://github.com/parcel-bundler/parcel/blob/dea272f0708fdabfe11f844f189df6421e7a3b65/packages/core/core/src/resolveOptions.js#L48

> nothing else, and it started working.

Yes, I can see how this fixed your problem. But there's not much we can do apart from documenting this behaviour.

**Comment by krnlde** [comment]
Thank you. I'll close the issue since the actual problem is resolved.
